iodata/formats/pdb.py

```diff
@@ -62,66 +60,116 @@ def load_one(lit: LineIterator) -> dict:
+    # 17 - 21       Integer        serial       Serial number of bonded atom
+    # 22 - 26       Integer        serial       Serial number of bonded atom
+    # 27 - 31       Integer        serial       Serial number of bonded atom
+    iatom0 = int(line[7:12]) - 1
+    for ipos in 12, 17, 22, 27:
+        serial_str = line[ipos: ipos + 5].strip()
+        if serial_str != "":
+            iatom1 = int(serial_str) - 1
+            if iatom1 > iatom0:
+                yield iatom0, iatom1
+
+
+@document_load_one("PDB", ['atcoords', 'atnums', 'atffparams', 'extra'], ['title', 'bonds'])
+def load_one(lit: LineIterator) -> dict:
+    """Do not edit this docstring. It will be overwritten."""
+    title = "PDB file from IOData"
     atnums = []
     attypes = []
     restypes = []
     chainids = []
     resnums = []
-    coords = []
+    atcoords = []
     occupancies = []
     bfactors = []
+    bonds = []
     molecule_found = False
     end_reached = False
-    title = "PDB file from IOData"
     while True:
         try:
             line = next(lit)
         except StopIteration:
             break
-        # If the PDB file has a title replace it.
+        # If the PDB file has a title, replace the default.
         if line.startswith("TITLE") or line.startswith("COMPND"):
             title = line[10:].rstrip()
         if line.startswith("ATOM") or line.startswith("HETATM"):
-            # get element symbol from position 77:78 in pdb format
-            words = line[76:78].split()
-            if not words:
-                # If not present, guess it from position 13:16 (atom name)
-                words = line[12:16].split()
-            # assign atomic number
-            symbol = words[0].title()
-            atnums.append(sym2num.get(symbol, sym2num.get(symbol[0], None)))
-            # atom name, residue name, chain id, & residue sequence number
-            attypes.append(line[12:16].strip())
-            restypes.append(line[17:20].strip())
-            chainids.append(line[21])
-            resnums.append(int(line[22:26]))
-            # add x, y, and z
-            coords.append([float(line[30:38]), float(line[38:46]), float(line[46:54])])
-            # get occupancies & temperature factor
-            occupancies.append(float(line[54:60]))
-            bfactors.append(float(line[60:66]))
+            (atnum, attype, restype, chainid, resnum, atcoord, occupancy,
+             bfactor) = _parse_pdb_atom_line(line)
+            atnums.append(atnum)
+            attypes.append(attype)
+            restypes.append(restype)
+            chainids.append(chainid)
+            resnums.append(resnum)
+            atcoords.append(atcoord)
+            occupancies.append(occupancy)
+            bfactors.append(bfactor)
             molecule_found = True
+        if line.startswith("CONECT"):
+            for iatom0, iatom1 in _parse_pdb_conect_line(line):
+                bonds.append([iatom0, iatom1, 1])
```

Comment:
Can you please clarify what `1` means here?  Is it meant to be bond-order? If so, what do you think of using `None` because it is not reported in PDB?

Reply:
I agree no bond type should be specified but with the current API, an integer must be provided, setting to `None` is not an option. The integers are "bond types" as defined in `iodata.periodic`. Setting it to `bond2num["un"]` (unknown) could be good temporary solution. My intention is to fix address this more generally with #222 (to introduce optional bond types and bond orders). With this PR and #248 I mainly wanted to get a better understanding of the different ways bonds are stored in various file formats. When addressing #222, this code will be updated.